### PR TITLE
[GuestConfiguration] GA (public) SDK 1.0.0

### PR DESCRIPTION
**Pull request by viveklingaiah**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
@vivlingaiah Please join the Azure org on github at [this](aka.ms/azuregithub) portal

**Comment by dsgouda** [comment]
Please link the corresponding REST spec PR too

**Comment by vivlingaiah** [comment]
spec link https://github.com/Azure/azure-rest-api-specs/blob/master/specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json

**Comment by vivlingaiah** [comment]
specs PR : https://github.com/Azure/azure-rest-api-specs/pull/4428

**Review by dsgouda — CHANGES_REQUESTED** [review]
While you are at it, please consider adding this ` <TreatWarningsAsErrors>true</TreatWarningsAsErrors>` [here](https://github.com/Azure/azure-sdk-for-net/blob/b03e6e5a71df49049126841ed783dbb5e4b95e7f/src/SDKs/GuestConfiguration/GuestConfiguration.Tests/GuestConfiguration.Tests.csproj#L4)
and fixing all the errors.

File: src/SDKs/GuestConfiguration/Management.GuestConfiguration/Microsoft.Azure.Management.GuestConfiguration.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.GuestConfiguration</PackageId>
     <Description>Provides Microsoft Azure Guest Configuration management operations including the ability to create, update and get guest configuration assignments, compliance statuses, guest configuration assignment reports.</Description>
-    <Version>0.10.0-preview</Version>
+    <Version>1.0.0</Version>
```
[the remark this answers is not in the document]
Comment (vivlingaiah) [reply src/SDKs/GuestConfiguration/Management.GuestConfiguration/Microsoft.Azure.Management.GuestConfiguration.csproj]: done

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge once CIs pass

**Pull request merged**